app/v1/hateoas/HateoasLinks.scala

```diff
@@ -28,13 +28,23 @@ trait HateoasLinks {
   private def metadataUri(appConfig: AppConfig, nino: String, calcId: String): String =
     baseUri(appConfig, nino) + s"/$calcId"
 
+  private def messagesUri(appConfig: AppConfig, nino: String, calcId: String): String =
+    baseUri(appConfig, nino) + s"/$calcId" + "/messages"
+
   // API resource links
   def trigger(appConfig: AppConfig, nino: String): Link =
     Link(href = baseUri(appConfig, nino), method = POST, rel = TRIGGER)
 
   def list(appConfig: AppConfig, nino: String): Link =
     Link(href = baseUri(appConfig, nino), method = GET, rel = SELF)
 
-  def getMetadata(appConfig: AppConfig, nino: String, calcId: String): Link =
-    Link(href = metadataUri(appConfig, nino, calcId), method = GET, rel = SELF)
+  def getMetadata(appConfig: AppConfig, nino: String, calcId: String, isSelf: Boolean = true): Link = isSelf match {
+    case true => Link (href = metadataUri (appConfig, nino, calcId), method = GET, rel = SELF)
```

Comment:
A plain if (isSelf) could be used here.

Also why not highlight what is changing by returning
```suggestion
    Link (href = metadataUri (appConfig, nino, calcId), method = GET, rel = if(isSelf) SELF else METADATA)
```